sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/query.sql

```diff
@@ -0,0 +1,113 @@
+DECLARE first_cohort_date DATE DEFAULT DATE(2024, 5, 1);
+
+DECLARE last_cohort_date DATE DEFAULT DATE_SUB(
+  CURRENT_DATE,
+  INTERVAL 8 DAY
+); -- want to make sure the last cohort we are reporting has had their full first 7 days
+WITH clients_first_seen AS (
+  SELECT
+    client_id,
+    first_seen_date,
+    country,
+    attribution_campaign,
+    attribution_content,
+    attribution_dltoken,
+    attribution_medium,
+    attribution_source
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_first_seen` --contains all new clients, including those that never sent a main ping
+  WHERE
+    first_seen_date
+    BETWEEN first_cohort_date
+    AND last_cohort_date
+),
+clients_last_seen AS (
+  SELECT
+    client_id,
+  -- note that this is the date we got each client's first main ping, which is potentially a different date from the first_seen_date above.
+    ANY_VALUE(first_seen_date) AS first_seen_date,
+  -- just the country value from their first main ping day
+    ANY_VALUE(CASE WHEN first_seen_date = submission_date THEN country END) AS country,
+  -- the date we would report on their conversion events (after we get data from their 7th day)
+    ANY_VALUE(
+      IF(submission_date = DATE_ADD(first_seen_date, INTERVAL 6 DAY), submission_date, NULL)
+    ) AS report_date,
+  -- their first week days of use, taken from their 7th day data
+    ANY_VALUE(
+      CASE
+        WHEN submission_date = DATE_ADD(first_seen_date, INTERVAL 6 DAY)
+          THEN BIT_COUNT(days_visited_1_uri_bits & days_interacted_bits)
+      END
+    ) AS dou,
+  -- if a client doesn't send a ping on `submission_date` their last active day's value will be carried forward
+  -- so we only take measurements from days that they send a ping.
+    SUM(
+      CASE
+        WHEN days_since_seen = 0
+          THEN COALESCE(active_hours_sum, 0)
+        ELSE 0
+      END
+    ) AS active_hours_sum,
+    SUM(
+      CASE
+        WHEN days_since_seen = 0
+          THEN COALESCE(search_with_ads_count_all, 0)
+        ELSE 0
+      END
+    ) AS search_with_ads_count_all
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_last_seen`
+  WHERE
+    submission_date >= first_cohort_date
+    AND submission_date
+    BETWEEN first_seen_date
+    AND DATE_ADD(first_seen_date, INTERVAL 6 DAY)
+  GROUP BY
+    client_id
+),
+combined AS (
+  SELECT
+    client_id,
+  -- we should use their actual first date as their cohort date
+    cfs.first_seen_date,
+    cfs.attribution_campaign,
+    cfs.attribution_content,
+    cfs.attribution_dltoken,
+    cfs.attribution_medium,
+    cfs.attribution_source,
+    cls.report_date,
```

Comment:
Talked about this in the meeting. I think we want to report a client as `FALSE` in the conversion events if they haven't sent a main ping (they have no rows from clients_last_seen, instrumented in the condition below as `cls.first_seen_date IS NULL`. We then write their `report_date` as their first_seen_date from CFS plus 27 days. 

```
COALESCE(cls.report_date, IF((cls.report_date IS NULL) AND (cls.first_seen_date IS NULL) AND (DATE_DIFF(CURRENT_DATE, cfs.first_seen_date, DAY) >= 27), DATE_ADD(cfs.first_seen_date, INTERVAL 27 DAY), NULL) AS report_date
```